scripts/2.TCGA-MLexample_Pathway.py

```diff
@@ -0,0 +1,535 @@
+cv_score_mat = pd.pivot_table(cv_score_df, values='score', index='l1_ratio', columns='alpha')
+ax = sns.heatmap(cv_score_mat, annot=True, fmt='.2%')
+ax.set_xlabel('Regularization strength multiplier (alpha)')
+ax.set_ylabel('Elastic net mixing parameter (l1_ratio)');
+
+
+# ## Use Optimal Hyperparameters to Output ROC Curve
+
+# In[39]:
+
+y_pred_train = pipeline.decision_function(X_train)
+y_pred_test = pipeline.decision_function(X_test)
+
+def get_threshold_metrics(y_true, y_pred, tissue='all'):
+    roc_columns = ['fpr', 'tpr', 'threshold']
+    roc_items = zip(roc_columns, roc_curve(y_true, y_pred))
+    roc_df = pd.DataFrame.from_items(roc_items)
+    auroc = roc_auc_score(y_true, y_pred)
+    return {'auroc': auroc, 'roc_df': roc_df, 'tissue': tissue}
+
+metrics_train = get_threshold_metrics(y_train, y_pred_train)
+metrics_test = get_threshold_metrics(y_test, y_pred_test)
+
+
+# In[40]:
+
+# Plot ROC
+plt.figure()
+for label, metrics in ('Training', metrics_train), ('Testing', metrics_test):
+    roc_df = metrics['roc_df']
+    plt.plot(roc_df.fpr, roc_df.tpr,
+        label='{} (AUROC = {:.1%})'.format(label, metrics['auroc']))
+plt.xlim([0.0, 1.0])
+plt.ylim([0.0, 1.05])
+plt.xlabel('False Positive Rate')
+plt.ylabel('True Positive Rate')
+plt.title('Predicting Hippo Signaling Pathway Aberations')
+plt.legend(loc='lower right');
+
+
+# ## Tissue specific performance
+
+# In[41]:
+
+tissue_metrics = {}
+for tissue in clinical.disease.unique():
+    sample_sub = clinical[clinical['disease'] == tissue].index.values
+    y_tissue_train = y_train[y_train.index.isin(sample_sub)]
+    y_tissue_pred_train = y_pred_train[y_train.index.isin(sample_sub)]
+    y_tissue_test = y_test[y_test.index.isin(sample_sub)]
+    y_tissue_pred_test = y_pred_test[y_test.index.isin(sample_sub)]
+
+    metrics_train = get_threshold_metrics(y_tissue_train, y_tissue_pred_train, tissue=tissue)
+    metrics_test = get_threshold_metrics(y_tissue_test, y_tissue_pred_test, tissue=tissue) 
+    
+    tissue_metrics[tissue] = [metrics_train, metrics_test]
+
+
+# In[42]:
+
+tissue_auroc = {}
+plt.figure()
+for tissue, metrics_val in tissue_metrics.items():
+    metrics_train, metrics_test = metrics_val
+    plt.subplot()
+    auroc = []
+    for label, metrics in ('Training', metrics_train), ('Testing', metrics_test):
+        roc_df = metrics['roc_df']
+        auroc.append(metrics['auroc'])
+        plt.plot(roc_df.fpr, roc_df.tpr,
+            label='{} (AUROC = {:.1%})'.format(label, metrics['auroc']))
+    tissue_auroc[tissue] = auroc
+    plt.xlim([0.0, 1.0])
+    plt.ylim([0.0, 1.05])
+    plt.xlabel('False Positive Rate')
+    plt.ylabel('True Positive Rate')
+    plt.title('Predicting Hippo Signaling in {}'.format(tissue))
+    plt.legend(loc='lower right'); 
+    plt.show()
+
+
+# In[43]:
+
+tissue_results = pd.DataFrame(tissue_auroc, index=['Train', 'Test']).T
+tissue_results = tissue_results.sort_values('Test', ascending=False)
+
+
+# In[44]:
```

Comment:
`<matplotlib.text.Text at 0x7f613946d0f0>` <-- such output prevents publication grade notebook. Perhaps fix by assigning to variable or including semicolon at end.